src/daemon/daemon.cpp

```diff
@@ -1968,7 +1968,17 @@ grpc::Status mp::Daemon::shutdown_vm(VirtualMachine& vm, const std::chrono::mill
     using St = VirtualMachine::State;
     const auto skip_states = {St::off, St::stopped, St::suspended};
 
-    if (std::none_of(cbegin(skip_states), cend(skip_states), [&state](const auto& st) { return state == st; }))
+    if (state == St::unknown)
+    {
+        if (delay.count() > 0)
+        {
+            return grpc::Status(grpc::StatusCode::FAILED_PRECONDITION,
+                                "Cannot delay shutdown when instance is in the unknown state");
```

Comment:
why?

Reply:
A delay of > 0 is a delayed shutdown, which we cannot do since we cannot get a valid ssh session into the instance.  My thinking is if someone is trying to do a delayed shutdown on an instance that is in an `UNKNOWN` state, then delayed shutdown is not valid.  I suppose the other option is to just kill the instance even when a delayed shutdown is requested, but that defeats the purpose of one wanting a delayed shutdown and would probably alarm the user.

Reply:
But if we made the ssh session optional in the delayed shutdown timer, it could work the same way, right? I mean, we would not need special behaviour for different states, other than printing or not printing wall messages depending on whether we can ssh into the machine (we keep the qemu interface in all cases)

Reply:
Changed